src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -282,30 +308,33 @@ private void walkScope(TypedVar symbol, final String namespace, boolean isDefaul
       }
       emit(getUnqualifiedName(symbol));
 
-      Function<ObjectType, String> relativeName = new Function<ObjectType, String>() {
-        @Override public String apply(ObjectType input) {
-          return !input.getDisplayName().startsWith(namespace) ? input.getDisplayName()
-              : input.getDisplayName().substring(namespace.length() + 1);
-        }
-      };
+      if (type.hasAnyTemplateTypes()) {
+        emit("<");
+        emit(FluentIterable.from(ftype.getTemplateTypeMap().getTemplateKeys()).transform(new Function<JSType, String>() {
```

Comment:
YMMV, but I'm not convinced this is substantially simpler than just using an explicit iterator:
```java
Iterator<JSTyle> it = ftype.getTemplateTypeMap().getTemplateKeys();
while (it.hasNext()) {
  emit(it.next().getDisplayName());
  if (it.hasNext()) emit(",");
}
```

That's ballpark the same SLOC, but works with a lot fewer concepts, doesn't yield the control flow to a fluent iterable so easier to step through, and is probably faster to boot (not that it matters).